relay-server/src/actors/project_cache.rs

```diff
@@ -570,11 +570,9 @@ impl ProjectCacheBroker {
         } = message;
 
         let project_cache = self.services.project_cache.clone();
-        self.get_or_create_project(project_key).update_state(
-            project_cache,
-            state.clone(),
-            no_cache,
-        );
+        let project = self.get_or_create_project(project_key);
+        project.update_state(project_cache, state.clone(), no_cache);
+        project.remove_expired_reservoir_rules();
```

Comment:
we check for expired reservoir rules every time we get a new project config. It's simply about cleaning up memory, having expired rules in the counters isn't an invalid state or anything.

Reply:
I would prefer if this was done internally in the `Project` rather than from the outside. It's already a bad pattern that the project cache needs to handle the dequeueing, but let's try to keep this here untouched and instead issue cleanup internally inside `update_state`.

Consequently the function `remove_expired_reservoir_rules` should become private.